server/setup-dev-server.sh

```diff
@@ -0,0 +1,59 @@
+while getopts c:k:d:g:r:p: flag
+do
+    case "${flag}" in
+        c) CERT_FILE=${OPTARG};;
+        k) KEY_FILE=${OPTARG};;
+        d) DEPLOY=${OPTARG};;
+        g) CERT_GENERATE=${OPTARG};;
+        r) RESET_DB=${OPTARG};;
+        p) PEM_FILE=${OPTARG};;
+    esac
+done
+
+DEPLOY="${DEPLOY:-1}"
+CERT_GENERATE="${CERT_GENERATE:-1}"
+PEM_FILE="${PEM_FILE:-~/.medperf.pem}"
+RESET_DB="${RESET_DB:-0}"
+echo $CERT_FILE
+echo $KEY_FILE
+echo $DEPLOY
+echo $CERT_GENERATE
+echo $RESET_DB
+
+if [ -z "$CERT_FILE" ]
+then
+  echo "CERT FILE must not be empty"
+  exit 1
+fi
+
+if [ -z "$KEY_FILE" ]
+then
+  echo "KEY FILE must not be empty"
+  exit 1
+fi
+
+if [ "$CERT_GENERATE" -eq 0 ]
+then
+  echo "No certs are generated as CERT_GENERATE flag is disabled"
+else
+    echo "Certs are generated"
+    openssl req -x509 -nodes -days 365 -newkey rsa:3072 -keyout cert.key -out cert.crt -subj "/C=US/ST=Any/L=Any/O=MedPerf/CN=127.0.0.1" -addext "subjectAltName=DNS:localhost,IP:127.0.0.1"
```

Comment:
Why do we need both crt and key file? We just need .crt file for client

`curl https://127.0.0.1:8000/ --cacert cert.crt `
`requests.request(url="https://127.0.0.1:8000/", method='get', verify='cert.crt')`

both work

Reply:
Is the validation failing for you even providing crt file?

Reply:
I wasn't aware requests could use the .crt file for verification. That was ignorance from my side, I'll make the required changes to pass the .crt instead of the .pem file I added

Reply:
@aristizabal95 , One question: Why can't we reuse the variable CERT_FILE var? Do we need to add extra parameter `CERT_PATH` as it is the same cert file?  In Line 41, cert.key and cert.crt needs to be changed to $CERT_FILE and $KEY_FILE respectively as well

With that, isn't this good enough to generate certs? `sh setup-dev-server.sh -c ~/.medperf_test.crt -k ~/.medperf_test.key`

Reply:
Agreed. Then we need to use the $CERT_FILE and $KEY_FILE inside the openssl command
```suggestion
    openssl req -x509 -nodes -days 365 -newkey rsa:3072 -keyout $KEY_FILE -out $CERT_FILE -subj "/C=US/ST=Any/L=Any/O=MedPerf/CN=127.0.0.1" -addext "subjectAltName=DNS:localhost,IP:127.0.0.1"
```
And we can remove the additional parameter and extra copy